### DISCUSSION TITLE
Childish achievements

### DISCUSSION THREAD

**Discussion by abelsromero** [community/community#18345] [Other Feature Feedback, Questions, & Ideas]
I understand this is opinionated and new achievements are meant to promote interactions with the platform.
But imho at the end of the day GitHub leans to a profesional platform where people work, new designs (yolo and shark?) are childish and not something I would like associated to me.

![image](https://user-images.githubusercontent.com/5781153/173181588-583e07ae-4271-4a30-8702-af99d8f105f1.png)

Plus, this will eventually be abused by people looking for attention and misinterpreted by others. We have enough proves of how the contributions tile is abused. For example people showing off hundreds of commits in a full green tile, and then on inspection it's all automated or posts to personal gh-pages that have no relationship with actual collab.

If this is pushed, at least make sure we can remove it and not be visible to others for users that consider it brand harmful, and even hide it for those that don't want to see it.

EDIT: I find several interesting topics and view in answers. But I still feel my point remains, many of the ones supporting it I think would be happy with just a stats view. No need to off icons, just show, number of reviews, PRs, etc in a useful way.

**Comment by ale-lit** [comment]
Hi! You can turn it off in the settings: https://github.com/settings/profile#profile-settings-heading

![image](https://user-images.githubusercontent.com/91823697/173181818-ab9f3a34-0115-41e0-84fd-cc314643d15d.png)

https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/showing-your-private-contributions-and-achievements-on-your-profile#changing-the-visibility-of-achievements

**Reply by milkcask** [reply]
it's quite obvious many of us want only the professional achievements on the profile.

**Reply by g-berthiaume** [reply]
I agree with @milkcask. 
Achievements could be a nice way of highlighting contributors' efforts.

Having your code land on mars is an achievement. Not creating a pull request.

**Reply by senseijack** [reply]
> Having your code land on mars is an achievement. Not creating a pull request.

I agree. If the badge only counted merged PRs for open source projects, or an organization that you don't belong to, it would be more meaningful. Right now it's silly, opening PRs against my organization's repos is my _job_, not an achievement.

**Reply by d4rkd0s** [reply]
`Default: off` might be preferred for features like this... 🤔 where the user profile appearance changes. 

As a lot of developers use their GitHub profile as a resume / CV.

So changing this with GitHub "feature drops" without the user knowing, could be frustrating, as I feel from the OP.

**Reply by GraysonMakesGames** [reply]
I agree, @milkcask, we usually only want professional achivements, and we should be able to hide certain achievements too.

**Reply by brentlintner** [reply]
You can't turn the gamey ones off, and keep the originals. Those felt like they mattered and were worth something.

**Reply by edmondburnett** [reply]
Exactly, it was fine when the achievements were rare, or related to science or the preservation of knowledge. Now it’s just getting messy and silly. This isn’t Instagram. 

Pretty typical of Microsoft to acquire something beloved and professional and turn it into a rattle toy.

**Comment by SakuraKoi** [comment]
Can't agree more, I want my profile to be professional and not childish. 
Arctic Code Vault and Mars Helicopter badge designs are both excellent, especially Arctic Code Vault Contributor, which is more beautifully detailed and I love it!
but these three... What's going wrong?
If github can't make something cool here, please at least provide a way to hide specific achievements from my profile, all or none is not a option.

**Reply by eureka-cpu** [reply]
Would also be great it achievements were more explicit in their description of what they represent, and how you can contribute to earn ones you would like on your profile.

**Comment by TSH3PH4NG** [comment]


**Reply by d4rkd0s** [reply]
Last checkbox at https://github.com/settings/profile

<img width="396" alt="image" src="https://user-images.githubusercontent.com/2348814/173923310-5e98f207-e8b8-4447-966f-f5eb4cc9229c.png">

**Reply by xyproto** [reply]
This removes all the achievements. Not ie. the "Yolo" achievement that can appear for just experimenting with a repository that nobody cares about.

**Comment by matt6frey** [comment]
Personally, I don't think there's anything wrong with them. They're just fun little aspects added to GitHub. They're not going to improve your CICD or give you a _professional_ gain but they're just gamifying developing and writing code or software. I take my work seriously, however, a little bit of horse play never hurts. I'm sure if there were some more incendiary badges that you'd never want, like I don't know, _tests never pass_ or _most rejected pull requests_ - sure, I think there'd be an issue. These are innocuous and meant to make you have a more enjoyable time while doing what you love.

**Comment by g-berthiaume** [comment]
I don't understand why you would try to gamify my work tool.
Do we give achievement to nurses when they take a patient's temperature?

**Reply by kai-burghardt** [reply]
I guess marketing geniuses of Micro$hit thought it was great.

**Comment by AymanLar** [comment]
i think there is nothing wrong with achievements, also the design could be more professional, like "YOLO" doesn't relate to merging a pull request without code review, its meaningless

**Comment by jhinter** [comment]
I'd love to see achievements reflecting how you help other developers. For instance: the number of code reviews conducted, and the number of comments in code reviews.

**Comment by apiraino** [comment]
+1 for removing this further gamification of the platform

**Comment by furio1999** [comment]
This particular achievement can be kept, but you have to check whether it was obtained in collaboration with another account of the same person. I was trying some CI/CD features with two accounts of mine and I received this badge!
Moreover, I suggest to look also at the relevance of repositories where the user contributed (number of stars, number of collaborators...)

**Comment by Iamgoofball** [comment]
I like achievements.

**Comment by Fabxx** [comment]
Would be nice to have a more professional redesign of the new achievements instead of removing them. It's nice to see that a user, especially newbies, get rewarded for their efforts (not counting in the ones who abuse doing PRs to just make the achivements appear).

**Comment by xyproto** [comment]
Please make it possible to select which achievements to include on the profile page. It doesn't make sense to me that this is an "all or nothing" option.

**Comment by lnpotter** [comment]
The achievements are pretty cool. Great work, it will definitely promote a more interactive environment between developers and their projects.

**Comment by brentlintner** [comment]
Things like Sponsors and Arctic contrib felt meaningful.

These feel like they are gamey, and I can't disable them and keep ones that I care about.

Just my thought as a user.

Cool stuff in general though! I just think there is a difference between the new ones. I have a random "Pull Shark" on my profile now, and it seems unprofessional and annoying.

**Comment by jlumbroso** [comment]
I don't understand some of the positions here: People think the Arctic contrib is more professional? But how did any of us participate in getting that badge? It was GitHub that took a random samples of repos, and if your repo happens to be in it, it means very little that is concrete.

In terms of the people complaining here: You are not the target audience. If you are already on GitHub, sufficiently passionate that you are engaging on GitHub's discussion forum, then you are in the top 0.01% of people on this Earth, who not only program, but also have extra time left to do programming-adjacent stuff like chat on a programming discussion board. Nobody needs to endorse your programming skills, because they are obvious.

But teaching in a university, when I have to advise students on development projects, the only source of truth is GitHub — any information I can get from their GitHub profile is useful, because everything else is usually a bit on the B.S. side — overstatements to the extent as to communicate useless information. It is very useful both for me, to evaluate the student; and for the student to be driven to learn about how to use GitHub, and how to contribute to open source, to have these badges to guide the discovery.

It's not a panacea, and if badges get any sort of interest, then there will be spamming as some clever people have pointed out. But overall, I think especially for novice developers, this is a good way of doing mentorship at scale.

**Reply by radar-blip** [reply]
I concur.

**Comment by kizdolf** [comment]
How is this turn ON by default ? yeah I YOLO some PR, but github YOLO some features apparently.

**Comment by GraysonMakesGames** [comment]
https://github.com/nathanielop/achievements
You were right, it's actually being abused 
(or at least, programs that can be used for abuse are being made)

I'll try to see if this works. If it does I'll edit this

Edit: tried using Cygwin to run shell scripts, does not work

**Comment by jfullstackdev** [comment]
fortunately we have options to hide the achievements and to make our profile private,

we know we have different opinions on this matter but all should be considered by GitHub, others really love this way, others don't 

even the term `childish` is very opinionated and will depend on the person saying it, like, if a certain adult is very annoyed by small things, that can be considered childish too, 

what maybe can be another request here is to have the option to select what badges will be displayed because as of now, it's either you display it or not, no chance to select, say only the Arctic badge
